minecraft/networking/connection.py

```diff
@@ -187,6 +187,15 @@ def write_packet(self, packet, force=False):
         else:
             self._outgoing_packet_queue.append(packet)
 
+    def listener(self, *packet_types, **kwds):
+        """
+        Shorthand decorator to register a function as a packet listener.
+        """
+        def _method_func(method):
+            self.register_packet_listener(method, *packet_types, **kwds)
```

Comment:
Oh, and: `_method_func` should end with `return method`, so it still gets properly bound at the site of use.